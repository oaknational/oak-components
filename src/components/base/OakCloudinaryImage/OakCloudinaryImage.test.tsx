import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { create } from "react-test-renderer";

import {
  OakCloudinaryConfigProvider,
  OakCloudinaryImage,
} from "./OakCloudinaryImage";

describe(OakCloudinaryImage, () => {
  it("matches snapshot", () => {
    const tree = create(
      <OakCloudinaryImage
        cloudinaryId="v1688316547/xqvnpdqx9u2awiykyb8s.jpg"
        alt="a test image"
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("respects the globally set config", () => {
    const { getByRole } = render(
      <OakCloudinaryImage
        cloudinaryId="v1608635688/image.jpg"
        alt="a test image"
      />,
    );

    expect(getByRole("img").getAttribute("src")).toMatch(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_3840/fl_keep_attribution/f_auto/q_auto/v1608635688/image",
    );
  });

  it("accepts a cloudinary image id", () => {
    const { getByRole } = render(
      <OakCloudinaryImage
        cloudinaryId="v1608635688/image.jpg"
        alt="a test image"
      />,
    );

    expect(getByRole("img").getAttribute("src")).toMatch(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_3840/fl_keep_attribution/f_auto/q_auto/v1608635688/image",
    );
  });

  it("accepts a full cloudinary url", () => {
    const { getByRole } = render(
      <OakCloudinaryImage
        cloudinaryId="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1608635688/image.jpg"
        alt="a test image"
      />,
    );

    expect(getByRole("img").getAttribute("src")).toMatch(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_3840/fl_keep_attribution/f_auto/q_auto/v1608635688/image",
    );
  });

  it("does not attempt to optimize the image when it is an SVG", () => {
    const { getByRole } = render(
      <OakCloudinaryImage
        cloudinaryId="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1608635688/image.svg"
        alt="a test image"
      />,
    );

    expect(getByRole("img").getAttribute("src")).toMatch(
      "https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1608635688/image",
    );
  });

  describe("private CDNs", () => {
    it("are respected when an id is passed", () => {
      const { getByRole } = render(
        <OakCloudinaryConfigProvider
          value={{
            cloud: {
              cloudName: "private-cloudinary-cloud",
            },
            url: {
              privateCdn: true,
            },
          }}
        >
          <OakCloudinaryImage
            cloudinaryId="https://private-cloudinary-cloud-res.cloudinary.com/image/upload/v1608635688/image.jpg"
            alt="a test image"
          />
        </OakCloudinaryConfigProvider>,
      );

      expect(getByRole("img").getAttribute("src")).toMatch(
        "https://private-cloudinary-cloud-res.cloudinary.com/image/upload/c_limit,w_3840/fl_keep_attribution/f_auto/q_auto/v1608635688/image.jpg",
      );
    });

    it("are respected when an full URL is passed", () => {
      const { getByRole } = render(
        <OakCloudinaryConfigProvider
          value={{
            cloud: {
              cloudName: "private-cloudinary-cloud",
            },
            url: {
              privateCdn: true,
            },
          }}
        >
          <OakCloudinaryImage
            cloudinaryId="https://private-cloudinary-cloud-res.cloudinary.com/image/upload/v1608635688/image.jpg"
            alt="a test image"
          />
        </OakCloudinaryConfigProvider>,
      );

      expect(getByRole("img").getAttribute("src")).toMatch(
        "https://private-cloudinary-cloud-res.cloudinary.com/image/upload/c_limit,w_3840/fl_keep_attribution/f_auto/q_auto/v1608635688/image.jpg",
      );
    });
  });
});
