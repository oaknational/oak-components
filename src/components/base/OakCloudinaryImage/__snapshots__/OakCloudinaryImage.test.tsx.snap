// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakCloudinaryImage matches snapshot 1`] = `
.c0 {
  position: relative;
  width: 100%;
  font-family: Lexend,sans-serif;
}

.c1 {
  object-fit: contain;
}

<div
  className="c0"
>
  <img
    alt="a test image"
    className="c1"
    data-nimg="fill"
    decoding="async"
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    sizes="100vw"
    src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_3840/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0"
    srcSet="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_640/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0 640w, https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_750/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0 750w, https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_828/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0 828w, https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_1080/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0 1080w, https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_1200/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0 1200w, https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_1920/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0 1920w, https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_2048/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0 2048w, https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/c_limit,w_3840/f_auto/q_auto/v1688316547/xqvnpdqx9u2awiykyb8s.jpg?_a=BAVAfVAO0 3840w"
    style={
      {
        "backgroundImage": "url("data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIyNjEiIGhlaWdodD0iMTUxIiBmaWxsPSJub25lIj48ZyBjbGlwLXBhdGg9InVybCgjYSkiPjxnIGNsaXAtcGF0aD0idXJsKCNiKSI+PGcgY2xpcC1wYXRoPSJ1cmwoI2MpIj48cGF0aCBmaWxsPSIjRTdGNkY1IiBkPSJNLTIwLTloMzAydjE3MC4wMTFILTIweiIvPjxwYXRoIGZpbGw9IiMyMjIiIGQ9Ik0xMjcuNzc5IDYzLjE4MWEyNy4xNDYgMjcuMTQ2IDAgMCAwLTMuOCAxLjYgMTYuNjk1IDE2LjY5NSAwIDAgMC03LjEgOC40YzAgLjEtLjEuMi0uMS4zLS43IDIuNC0uNiAyIDEuMyAyLjMgMS45LjMgMSAuNSAxIDEuMy0uMSA4LjggNC4xIDE1LjEgMTEuNCAxOS42YTEuNDk4IDEuNDk4IDAgMCAwIDEuNy4yYzUuNy0yLjYgOS4zLTcgMTAuMy0xMy4yYTEuMDAxIDEuMDAxIDAgMCAxIDEtMWwzLS4yYy44IDAgMS4zLjIgMS4yIDEuMmExNy45MDggMTcuOTA4IDAgMCAxLTMuMiA5LjIgMjMuNyAyMy43IDAgMCAxLTEwLjkgOS4xIDUuNDA2IDUuNDA2IDAgMCAxLTQuNS0uMiAyNi4yOSAyNi4yOSAwIDAgMS04LjUtNi42IDI1Ljg4OSAyNS44ODkgMCAwIDEtNi40LTE0LjRjMC0uNi0uMi0uNy0uOC0uOC0yLjUtLjQtMi41LS4xLTIuMy0yLjlhMTkuMjkgMTkuMjkgMCAwIDEgMTAuOC0xNi42IDM4Ljk1MyAzOC45NTMgMCAwIDEgNS43LTIuMWMuNDU3LS4zLjc4LS43NjYuOS0xLjNhMTQuMDk1IDE0LjA5NSAwIDAgMSAzLjUtNi4zbC4zLS4zYzEuOS0yIDIuNi0yIDQuMy4ybC40LjVjMS4xIDEuMSAxIDEuNS0uMSAyLjZhMTEuODgzIDExLjg4MyAwIDAgMC0zLjIgNC40YzIuNjQ1LjE2OCA1LjIxNS45NTYgNy41IDIuMyA1LjcgMy41IDkuMiA4LjMgOS45IDE1IC4wMTYuOTAxLS4wMTcgMS44MDMtLjEgMi43IDAgLjgtLjYgMS0xLjIgMS4yYTE2LjEgMTYuMSAwIDAgMS0xMS0uNyAxNy45IDE3LjkgMCAwIDEtMTAuOS0xMy42IDkuNzQyIDkuNzQyIDAgMCAxLS4xLTEuOVptMTguMSAxMi4yYy40LTUuNS02LjktMTIuNi0xMy0xMi4xLjUgNi41IDcuNiAxMi44IDEzIDEyLjFaIiBvcGFjaXR5PSIuMSIvPjwvZz48L2c+PC9nPjxkZWZzPjxjbGlwUGF0aCBpZD0iYSI+PHBhdGggZmlsbD0iI2ZmZiIgZD0iTTAgMGgyNjF2MTUxSDB6Ii8+PC9jbGlwUGF0aD48Y2xpcFBhdGggaWQ9ImIiPjxwYXRoIGZpbGw9IiNmZmYiIGQ9Ik0tMjAtOWgzMDJ2MTcwLjAxMUgtMjB6Ii8+PC9jbGlwUGF0aD48Y2xpcFBhdGggaWQ9ImMiPjxwYXRoIGZpbGw9IiNmZmYiIGQ9Ik0tMjAtOWgzMDJ2MTcwLjAxMUgtMjB6Ii8+PC9jbGlwUGF0aD48L2RlZnM+PC9zdmc+")",
        "backgroundPosition": "50% 50%",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "bottom": 0,
        "color": "transparent",
        "height": "100%",
        "left": 0,
        "objectFit": undefined,
        "objectPosition": undefined,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": "100%",
      }
    }
  />
</div>
`;
